From b5448e6d8dab7165fa800e1f79bdae09bb0c5310 Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Fri, 24 Jan 2020 11:48:06 +0200
Subject: [PATCH] Add alarl target only if building repc

Fails on cross-compilation with error:
Cannot specify sources for target "repoc" which is not built by this project.

Change-Id: I0fb757d7d9f599dd6dc7cddcabbf048ddf97a599
---
 tools/repc/CMakeLists.txt | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/tools/repc/CMakeLists.txt b/tools/repc/CMakeLists.txt
index bc34e22..06a3f6e 100644
--- a/tools/repc/CMakeLists.txt
+++ b/tools/repc/CMakeLists.txt
@@ -31,12 +31,14 @@ qt_add_tool(repc
         Qt::CorePrivate
 )
 
-# QLALR Grammars:
-qt_process_qlalr(
-    repc
-    ${CMAKE_SOURCE_DIR}/src/repparser/parser.g
-    ""
-)
+if(QT_WILL_BUILD_TOOLS)
+    # QLALR Grammars:
+    qt_process_qlalr(
+        repc
+        ${CMAKE_SOURCE_DIR}/src/repparser/parser.g
+        ""
+    )
+endif()
 
 #### Keys ignored in scope 1:.:.:repc.pro:<TRUE>:
 # QMAKE_TARGET_DESCRIPTION = "Qt Remote Objects Compiler"
